File: backend/src/main/java/com/tripfriend/domain/recruit/recruit/dto/RecruitListResponseDto.java
//package com.tripfriend.domain.recruit.recruit.dto;
//
//import com.fasterxml.jackson.annotation.JsonProperty;
//import com.tripfriend.domain.member.member.entity.Gender;
//import com.tripfriend.domain.recruit.recruit.entity.Recruit;
//import lombok.AllArgsConstructor;
//import lombok.Builder;
//import lombok.Data;
//import lombok.NoArgsConstructor;
//
//import java.time.LocalDate;
//import java.time.LocalDateTime;
//
//@Data
//@Builder
//@NoArgsConstructor
//@AllArgsConstructor
//public class RecruitListResponseDto {
//    private Long recruitId;
//    //    private Member member;
//    private String memberProfileImage;
//    private String memberNickname;
//    private String genderRestriction;
//    private String ageRestriction;
//    //    private List<Apply> applies;
////    private Place place;
//    private String placeCityName;
//    private String placePlaceName;
//    private String title;
//    //    private String content;
//    @JsonProperty("isClosed")
//    private boolean isClosed;
//    private LocalDate startDate;
//    private LocalDate endDate;
//    private String travelStyle;
//    //    private boolean sameGender;
////    private boolean sameAge;
//    private Integer budget = 0;
//    private Integer groupSize = 2;
//    private LocalDateTime createdAt;
//    private LocalDateTime updatedAt;
//
//    public RecruitListResponseDto(Recruit recruit){
//        this.recruitId = recruit.getRecruitId();
////        this.member = recruit.getMember();
//        this.memberNickname = recruit.getMember().getNickname();
//        this.memberProfileImage = recruit.getMember().getProfileImage();
////        this.applies = recruit.getApplies();
////        this.place = recruit.getPlace();
//        this.placeCityName = recruit.getPlace().getCityName();
//        this.placePlaceName = recruit.getPlace().getPlaceName();
//        this.title = recruit.getTitle();
////        this.content = recruit.getContent();
//        this.isClosed = recruit.isClosed();
//        this.startDate = recruit.getStartDate();
//        this.endDate = recruit.getEndDate();
//        this.travelStyle = recruit.getTravelStyle().getKoreanName();
////        this.sameGender = recruit.isSameGender();
////        this.sameAge = recruit.isSameAge();
//        this.budget = recruit.getBudget();
//        this.groupSize = recruit.getGroupSize();
//        this.createdAt = recruit.getCreatedAt();
//        this.updatedAt = recruit.getUpdatedAt();
//        // 성별 제한 설정
//        this.genderRestriction = recruit.isSameGender() && recruit.getMember().getGender() != Gender.UNKNOWN
//                ? (recruit.getMember().getGender() == Gender.MALE ? "남자만" : "여자만")
//                : (recruit.getMember().getGender() == Gender.UNKNOWN ? "알 수 없음" : "모든 성별");
//
//        // 나이대 제한 설정
//        this.ageRestriction = recruit.isSameAge()
//                ? switch (recruit.getMember().getAgeRange()) {
//            case TEENS -> "10대만";
//            case TWENTIES -> "20대만";
//            case THIRTIES -> "30대만";
//            case FORTIES_PLUS -> "40대 이상만";
//            default -> "알 수 없음";
//        }
//                : "모든 연령대";
//    }
//}

File: backend/src/main/java/com/tripfriend/domain/recruit/recruit/dto/RecruitListResponseDto.kt
package com.tripfriend.domain.recruit.recruit.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.tripfriend.domain.member.member.entity.Gender
import com.tripfriend.domain.recruit.recruit.entity.Recruit
import java.time.LocalDate
import java.time.LocalDateTime

data class RecruitListResponseDto(
    val recruitId: Long,
    val memberProfileImage: String?,
    val memberNickname: String?,
    val genderRestriction: String,
    val ageRestriction: String,
    val placeCityName: String?,
    val placePlaceName: String?,
    val title: String,
    @JsonProperty("isClosed")
    val isClosed: Boolean,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val travelStyle: String,
    val budget: Int = 0,
    val groupSize: Int = 2,
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?
) {
    constructor(recruit: Recruit) : this(
        recruitId = recruit.recruitId!!,
        memberProfileImage = recruit.member.profileImage,
        memberNickname = recruit.member.nickname,
        genderRestriction = if (recruit.sameGender && recruit.member.gender != Gender.UNKNOWN) {
            if (recruit.member.gender == Gender.MALE) "남자만" else "여자만"
        } else {
            if (recruit.member.gender == Gender.UNKNOWN) "알 수 없음" else "모든 성별"
        },
        ageRestriction = if (recruit.sameAge) {
            when (recruit.member.ageRange) {
                com.tripfriend.domain.member.member.entity.AgeRange.TEENS -> "10대만"
                com.tripfriend.domain.member.member.entity.AgeRange.TWENTIES -> "20대만"
                com.tripfriend.domain.member.member.entity.AgeRange.THIRTIES -> "30대만"
                com.tripfriend.domain.member.member.entity.AgeRange.FORTIES_PLUS -> "40대 이상만"
                else -> "알 수 없음"
            }
        } else {
            "모든 연령대"
        },
        placeCityName = recruit.place.cityName,
        placePlaceName = recruit.place.placeName,
        title = recruit.title,
        isClosed = recruit.isClosed,
        startDate = recruit.startDate,
        endDate = recruit.endDate,
        travelStyle = recruit.travelStyle.koreanName,
        budget = recruit.budget,
        groupSize = recruit.groupSize,
        createdAt = recruit.createdAt,
        updatedAt = recruit.updatedAt
    )
}
